odoo/addons/base/transpiler/js_transpiler.py

```diff
@@ -0,0 +1,594 @@
+        new_object = matchobj["object"].replace(" as ", ": ")
+        return f"{matchobj['space']}const {new_object} = require({matchobj['path']})"
+    return IMPORT_BASIC_RE.sub(repl, content)
+
+
+IMPORT_LEGACY_DEFAULT_RE = re.compile(r"""
+    ^
+    (?P<space>\s*)                                      # space and empty line
+    import\s+                                           # import
+    (?P<identifier>\w+)\s*                              # default variable name
+    from\s*                                             # from
+    (?P<path>(\"\w+\.\w+\")|('\w+\.\w+')|(`\w+\.\w+`))  # legacy alias file ("addon_name.module_name")
+    """, re.MULTILINE | re.VERBOSE)
+
+
+def convert_legacy_default_import(content):
+    """
+    Transpile legacy imports (that were used as they were default import).
+    Legacy imports means that their name is not a path but a <addon_name>.<module_name>.
+    It requires slightly different processing.
+
+    .. code-block:: javascript
+
+        // before
+        import module_name from "addon.module_name"
+        // after
+        const module_name = require("addon.module_name")
+    """
+    repl = r"""\g<space>const \g<identifier> = require(\g<path>)"""
+    return IMPORT_LEGACY_DEFAULT_RE.sub(repl, content)
+
+
+IMPORT_DEFAULT = re.compile(r"""
+    ^
+    (?P<space>\s*)                      # space and empty line
+    import\s+                           # import
+    (?P<identifier>\w+)\s*              # default variable name
+    from\s*                             # from
+    (?P<path>(\".*\")|('.*')|(`.*`))    # "file path" ("some/path")
+    """, re.MULTILINE | re.VERBOSE)
+
+
+def convert_default_import(content):
+    """
+    Transpile the default import call.
+
+    .. code-block:: javascript
+
+        // before
+        import something from "some/path"
+        // after
+        const something = require("some/path")[Symbol.for("default")]
+    """
+    repl = r"""\g<space>const \g<identifier> = require(\g<path>)[Symbol.for("default")]"""
+    return IMPORT_DEFAULT.sub(repl, content)
+
+
+RELATIVE_REQUIRE_RE = re.compile(r"""
+    require\((["'])([^@\"\']+)(["'])\) # require("some/path")
+    """, re.VERBOSE)
+
+
+def convert_relative_require(url, content):
+    """
+    Convert the relative path contained in a 'require()'
+    to the new path system (@module/path)
+    .. code-block:: javascript
+
+        // Relative path:
+        // before
+        require("./path")
+        // after
+        require("@module/path")
+
+        // Not a relative path:
+        // before
+        require("other_alias")
+        // after
+        require("other_alias")
+    """
+    new_content = content
+    for open_quote, path, close_quote in RELATIVE_REQUIRE_RE.findall(new_content):
+        if path.startswith(".") and "/" in path:
+            new_content = re.sub(rf"require\({str(open_quote)}{path}{str(close_quote)}\)", f'require("{relative_path_to_module_path(url, path)}")', new_content)
```

Comment:
If I am not mistaken, both `open_quote` and `close_quote` are strings.  So why `str()` them?
```suggestion
            pattern = rf"require\({open_quote}{path}{close_quote}\)"
            repl = f'require("{relative_path_to_module_path(url, path)}")'
            new_content = re.sub(pattern, repl, new_content)
```

Reply:
The f-string would stringify them anyway.